rosbag2_transport/src/rosbag2_transport/recorder.hpp

```diff
@@ -42,10 +47,21 @@ class Recorder
 private:
   std::shared_ptr<GenericSubscription> create_subscription(
     const std::string & topic_name, const std::string & topic_type);
+  std::future<void> launch_topics_discovery(
+    std::chrono::milliseconds topic_polling_interval,
+    bool is_discovery_disabled,
+    const std::vector<std::string> & topics_to_record = {});
+  void subscribe_all_missing_topics(
+    const std::unordered_map<std::string, std::string> & all_topics_and_types);
+  void subscribe_topic(const rosbag2::TopicMetadata & topic);
+  bool is_every_topic_subscribed(const std::vector<std::string> & topics_to_record) const;
+  void record_messages() const;
 
   std::shared_ptr<rosbag2::Writer> writer_;
   std::shared_ptr<Rosbag2Node> node_;
   std::vector<std::shared_ptr<GenericSubscription>> subscriptions_;
+  std::vector<std::string> subscribed_topics_;
```

Comment:
i believe an `unordered_set<std::string>` makes more sense here, because you really just need a quick lookup whether it's subscribed or not.